src/v/raft/prevote_stm.cc

```diff
@@ -61,6 +61,7 @@ prevote_stm::process_reply(vnode n, ss::future<result<vote_reply>> f) {
     try {
         if (_prevote_timeout < clock_type::now()) {
             vlog(_ctxlog.trace, "prevote ack from {} timed out", n);
+            (void)f.get0(); // rethrow if exceptional, so we handle it below
```

Comment:
i think we can move waiting for future out of an `if` statement. The request is already protected with timeout so checking the timeout here is redundant. 
So it would be something like:
```c++
try {
  auto r = f.get();
  it(...){
...
} catch(...) {
...
}
```

Reply:
just to clarify here @mmaslankaprv 

> can move waiting for future out of an if statement

we aren't waiting, right? the future is immediately available at this point?

> checking the timeout here is redundant.

are these few cases already handled by the caller?

```
            voter_reply->second._is_failed = true;
            voter_reply->second._is_pending = false;
```